### PR TITLE
Mark BackdropFilter as inheriting opacity, enabling fading of BDFs

### PR DESCRIPTION
**Pull request by flar**
Customers have complained about BackdropFilter not working well with an enclosing OpacityLayer. This led to a bug in iOS dialogs from a few years back (see https://github.com/flutter/flutter/issues/31706).

While this PR does not solve all cases where a BackdropFilter was nullified by being inside an enclosing saveLayer, it does solve a common case where the BackdropFilter is a direct (and only) child of an OpacityLayer.

Basically the fix is to mark the BackdropFilterLayer as able to inherit opacity which allows it to combine the opacity of a parent OpacityLayer into its own saveLayer.

Some refactoring of the AutoCachePaint facility was involved to more easily enable sharing of the object for both DisplayList and SkCanvas paths in the layers (as the 2 rendering variants of BackdropFilterLayer now need to do).

### CODE REVIEW

**Comment by flar** [comment]
CC @antholeole who was running into this issue recently

**Comment by antholeole** [comment]
I can confirm that this unblocks me! 

Before:
https://user-images.githubusercontent.com/48811365/176983029-9666e3be-4ec5-4c55-a024-b40217c7f18a.mov

After:
https://user-images.githubusercontent.com/48811365/176983047-95718301-66cf-4e9f-b888-4708dcff80f5.mov

 Thanks for the fix! This is awesome :)

**Review by bdero — APPROVED** [review]

**Comment by bdero** [comment]
Haven't looked at the log spam mentioned above, but setting opacity on BDF layers is fine in Impeller.

**Comment by flar** [comment]
> There's a bit of logspam, not sure if this is because I pulled your branch first and merged this over:
> 
> ```
> [VERBOSE-2:backdrop_filter_layer.cc(58)] Inherited opacity: 0x70000a6b8f30
> [VERBOSE-2:backdrop_filter_layer.cc(58)] Inherited opacity: 0x70000a6b8f30
> [VERBOSE-2:backdrop_filter_layer.cc(58)] Inherited opacity: 0x70000a6b8f30
> ... (~20x)
> ```

Thanks for checking! There should be a pre-check that maybe doesn't reject pushes, but indicates if there are new calls to FML_LOG in them.
